### PR TITLE
Capsule MeshBuilder

### PR DESCRIPTION
**Pull request by Pryme8**
I still need to do linting and comments.

Will do ASAP

### CODE REVIEW

File: src/Meshes/meshBuilder.ts [hunk 1]
```diff
@@ -546,4 +547,36 @@ export class MeshBuilder {
     public static CreateDecal(name: string, sourceMesh: AbstractMesh, options: { position?: Vector3, normal?: Vector3, size?: Vector3, angle?: number }): Mesh {
         return DecalBuilder.CreateDecal(name, sourceMesh, options);
     }
+
+
+    /**
+     * Creates a decal mesh.
```
Comment (deltakosh) [inline src/Meshes/meshBuilder.ts:553]: wrong doc ;)

**Comment by deltakosh** [comment]
Can you also provide a visual test for it?
File to update: https://github.com/BabylonJS/Babylon.js/blob/master/tests/validation/config.json
Image will be saved here: https://github.com/BabylonJS/Babylon.js/tree/master/tests/validation/ReferenceImages

File: src/Meshes/meshBuilder.ts (hunk 1 above)
Comment (Pryme8) [reply src/Meshes/meshBuilder.ts:553]: Ill fix all that asap!  I had no time to do docs yet.

**Comment by Pryme8** [comment]
Yes I will do tests.  I had this code sitting around forever and just stumbled across it right as @MackeyK24 was asking for it so figured might as well bust it out.

**Comment by Pryme8** [comment]
so I use our tools to save a screenshot of the scene from a playground and then just update that json and upload the image?

**Comment by sebavan** [comment]
Yup exactly :-) create the test, run the tool, save the screenshot in the tests/validation/ReferenceImages folder and you should be good.

**Comment by Pryme8** [comment]
I'll try to get this resolved today.

**Comment by Pryme8** [comment]
Powers back on from the fires so I am on this now.

**Comment by Pryme8** [comment]
> Yup exactly :-) create the test, run the tool, save the screenshot in the tests/validation/ReferenceImages folder and you should be good.

What size do I make the image?

**Comment by Pryme8** [comment]
@sebavan I keep getting a image that is not the same dimensions as the other images it seems.

BABYLON.Tools.CreateScreenshot(engine, camera, {width:800, height:600})  is all I figured I needed to do, but that is not making my image match.  Is there a step I am missing?

**Comment by Pryme8** [comment]
I need some help with the testing stuff, I think I almost have it just not sure though what all the steps really are or what the expectation is.  >_<.

File: src/Meshes/Builders/capsuleBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,312 @@
+import { VertexData } from "../mesh.vertexData";
+import { Vector2, Vector3 } from "../../Maths/math.vector";
+import { Mesh, _CreationDataStorage } from "../mesh";
+    /**
```
Comment (deltakosh) [inline src/Meshes/Builders/capsuleBuilder.ts:4]: tabs seem weird

File: tests/validation/config.json [hunk 3]
```diff
@@ -332,6 +332,11 @@
             "referenceImage": "lines.png",
             "replace": "updateVerticesDataDirectly, updateVerticesData"
         },
+        {
+            "title": "Capsule",
+            "playgroundId": "#JAFIIU",
```
Comment (deltakosh) [inline tests/validation/config.json:337]: all good!

**Review by deltakosh — APPROVED** [review]

File: src/Meshes/Builders/capsuleBuilder.ts (hunk 2 above)
Comment (Pryme8) [reply src/Meshes/Builders/capsuleBuilder.ts:4]: will fix that right now!

**Comment by deltakosh** [comment]
Ok..
I'm gonna take care of it

**Comment by deltakosh** [comment]
All of this can be fixed by running npm run build locally...

**Comment by Pryme8** [comment]
I don't understand, I run that before I ever push it every time.  A lot of the time the only error I will get is a visual test wont work.  Sometimes it tells me I need to add comments and that's it..  that is driving me crazy.

**Comment by deltakosh** [comment]
" I will get is a visual test wont work." this is the problem. Working on fixing it right now

**Comment by deltakosh** [comment]
well it is quite easy for this one:
image

**Comment by Pryme8** [comment]
ooooohhhh, I wonder if what is happening is sometimes I accidentally highlight one of the files I have modified in VScode after the build and discard the changes.

I will have to pay more attention because that should have had a #0 behind it if it was the most recent file that I had.

Also was not really ready for it to be merged, I had orientation stuff to fix.   I will in the future wait to make a PR till its more in a ready state, this was kinda rushed sorry.

**Comment by deltakosh** [comment]
Feel free to send an additional PR. no worries

**Comment by Popov72** [comment]
> Also was not really ready for it to be merged

The best thing to do is to create your PG as Draft, this way it can't be merged. Once everyone agrees that the PR is ok, you can set it as a regular PR and remove the "draft" tag.

That's what I'm doing now when my PR is not a trivial one.

**Comment by MackeyK24** [comment]
How do you use... What are the required options ??

**Comment by Pryme8** [comment]
Im doing documents on all this soon

**Pull request merged**